go-controller/pkg/cni/types.go

```diff
@@ -79,6 +81,32 @@ type Response struct {
 	KubeAuth  *KubeAPIAuth
 }
 
+func (response *Response) Marshal() ([]byte, error) {
+	return json.Marshal(response)
+}
+
+// Filter out kubeAuth, since it might contain sensitive information.
+func (response *Response) MarshalForLogging() ([]byte, error) {
+	var noAuthJSON []byte
+	var err error
+
+	if response == nil {
+		return nil, nil
+	}
+
+	noAuth := struct {
+		Result    *current.Result
+		PodIFInfo *PodInterfaceInfo
+	}{Result: response.Result, PodIFInfo: response.PodIFInfo}
```

Comment:
if you look at the details of `cmdAdd`, every `Response` has only one of `Result` and `PodIFInfo` set. So you could just marshal and log whichever of them is non-`nil`

Reply:
That's also true! I just pushed a commit with the changes you requested.